modules/videoio/cmake/detect_ueye.cmake

```diff
@@ -0,0 +1,27 @@
+if(NOT HAVE_UEYE)
+  if(WIN32)
+    if(X86_64)
+      set(_WIN_LIB_SUFFIX "_64")
+    endif()
+  endif()
+  find_path(UEYE_INCLUDE "ueye.h"
+    PATHS "${UEYE_ROOT}" ENV UEYE_ROOT "/usr" "C:/Program Files/IDS/uEye/Develop"
+    HINTS "${regpath}"
+    PATH_SUFFIXES "include")
+  find_library(UEYE_LIBRARY ueye_api${_WIN_LIB_SUFFIX}
+    PATHS "${UEYE_ROOT}" ENV UEYE_ROOT "/usr" "C:/Program Files/IDS/uEye/Develop"
+    HINTS "${regpath}"
+    PATH_SUFFIXES "lib")
+  if(UEYE_INCLUDE AND UEYE_LIBRARY)
+    set(HAVE_UEYE TRUE)
+  else()
+    message(FATAL_ERROR "INC ${UEYE_INCLUDE} LIB ${UEYE_LIBRARY}")
```

Comment:
> FATAL_ERROR

We don't use that during dependencies search.
`HAVE_UEYE` contains result.
Consider using STATUS instead with more clear message.